sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -49,6 +51,9 @@ func managedFieldsTransformer(obj interface{}) (interface{}, error) {
 }
 
 func (k *listenerImpl) handleAllEvents() {
+	k.contextMtx.Lock()
+	defer k.contextMtx.Unlock()
```

Comment:
Wow, this will be locked for entire func that is not small... not saying it is bad, but I hope there are no deadlocks.

Reply:
Could we maybe copy the reference to the context here to not have to block during the entire function?

Reply:
It's fine. This is only called when starting up the listener. This is just to avoid a single case where sensor restarts _before_ the previous reconciliation finished. This lock will prevent the second start process (with a new context) to conflict with the first one and cause a data race (swap the context pointer while its being used to create the handlers).

Reply:
> Could we maybe copy the reference to the context here to not have to block during the entire function?

I don't understand this suggestion